### DISCUSSION TITLE
Multiple entities, not clear on how to add the correct param

### DISCUSSION THREAD

**Discussion by shoham-utila** [tngan/samlify#482] [Q&A]
I want to build a tenantable SSO login with SAML, which means I'll have a provider for each tenant, and I wanted to use the multiple entities approach from the docs, but there is something I'm unclear on, in the example they enter the IdP type as a request param, but it doesn't show how I create each login request to return to a diffrent url (the same url but with a different param) so I could identify the correct IdP
I'm adding the code from the docs with my comments to explain

```
// define SP
const sp = saml.ServiceProvider({
  metadata: fs.readFileSync('./metadata_sp.xml')
});
// define multiple IdPs
const defaultIdP = saml.IdentityProvider({
  metadata: fs.readFileSync('./metadata_idp_default.xml')
});
const oneloginIdP = saml.IdentityProvider({
  metadata: fs.readFileSync('./metadata_onelogin.xml')
});
// URL routing for SP-initiated SSO 
router.get('/spinitsso-post/:idp', (req, res) => { // how do I make the IdP to redirect to this address with a param I want?
  let targetIdP = undefined;
  switch(req.params.idp) {
    case 'onelogin': {
      targetIdP = oneloginIdP;
      break;
    }
    default: {
      targetIdP = idp;
      break;
    }
  }
  return sp.createLoginRequest(targetIdP, 'post', (req, res) => res.render('actions', req));
});
```

**Comment by tngan** [comment]
@bsShoham It's determined via the mapping of url parameters, for example, if client side accesses `/spinitsso-post/onelogin`, it will take the `oneloginIdP` as the target. `sp.createLoginRequest` only returns you the context of the request, the redirect action you need to handle yourself.

The redirect url returned from the create function should follow the sso entry point specified in the onelogin metadata.
